ocaml/xapi-aux/pool_role.ml

```diff
@@ -50,11 +50,11 @@ let read_pool_role () =
   try
     let s = Astring.String.trim
         (Unixext.string_of_file !Constants.pool_config_file) in
-    match Astring.String.cut ~sep:":" s with
-    | Some ( "master", "")  -> Master
-    | Some ( "slave", m_ip) -> Slave m_ip
-    | Some ( "broken", "")  -> Broken
-    | _ -> failwith "cannot parse pool_role from pool config file"
+    match Astring.String.cuts ~sep:":" s with
+    | ["master"]      -> Master
+    | ["slave"; m_ip] -> Slave m_ip
```

Comment:
Could you use Slave (String.concat ":" m_ip), just in case? It's unfortunate astring doesn't have a direct replacement to the split with limit.

Reply:
yes, for ipv6 (NB. this was not the original behaviour)